File: dsa-old/data_structures/LinearSearch/LinearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int[], int, int);

int main()
{
    int n;
    cout << "Enter size of array: ";
    cin >> n;

    int arr[n];
    cout << "Enter elements: ";
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    int key;
    cout << "Enter key: ";
    cin >> key;

    int foundAt = linearSearch(arr, n, key);
    (foundAt == -1) ? cout << "Element not found" : cout << "Element found at: " << foundAt << " [0-indexing]" << endl;
}

int linearSearch(int arr[], int n, int key)
{
    for (int i = 0; i < n; i++)
    {
        if (key == arr[i])
        {
            return i;
        }
    }

    return -1;
}

File: dsa/datastucures/arrays/LinearSearch.java
import java.util.Scanner;

class LinearSearch {
  public static void main(String[] args) {

    System.out.print("Enter size: ");    
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();


    int[] arr = new int[n];

    System.out.print("Enter array: ");
    for (int i = 0; i < arr.length; i++) {
      arr[i] = sc.nextInt();
    }    

    System.out.println("Enter element to be searched: ");    
    int search = sc.nextInt();

    for (int i = 0; i < arr.length; i++) {
      if (arr[i] == search) {
        System.out.println("Found at loc: " + i);
      }
    }
  }
}